tests/DocumentationExamplesTest.php

```diff
@@ -1408,6 +1410,57 @@ public function testTransactions_retry_example_3()
         }
     }
 
+    function testCausalConsistency()
+    {
+        $this->skipIfCausalConsistencyIsNotSupported();
+
+        // Prep
+        $client = new Client($this->getUri());
+        $test = $client->selectDatabase(
+            'test',
+            [ 'writeConcern' => new \MongoDB\Driver\WriteConcern(\MongoDB\Driver\WriteConcern::MAJORITY) ]
+        );
+        $items = $client->test->items;
+
+        $items->drop();
+        $items->insertOne(
+            [ 'sku' => '111', 'name' => 'Peanuts', 'start' => new \MongoDB\BSON\UTCDateTime() ]
+        );
+
+        // Start Causal Consistency Example 1
+        $s1 = $client->startSession([ 'causalConsistency' => true ]);
+
+        $items->updateOne(
+            [ 'sku' => '111', 'end' => [ '$exists' => false ] ],
+            [ '$currentDate' => [ 'end' => true ] ],
+            [ 'session' => $s1 ]
+        );
+        $items->insertOne(
+            [ 'sku' => '111-nuts', 'name' => 'Pecans', 'start' => new \MongoDB\BSON\UTCDateTime() ],
+            [ 'session' => $s1 ]
+        );
+        // End Causal Consistency Example 1
+
+        ob_start();
+
+        // Start Causal Consistency Example 2
+        $s2 = $client->startSession([ 'causalConsistency' => true ]);
+        $s2->advanceClusterTime($s1->getClusterTime());
+        $s2->advanceOperationTime($s1->getOperationTime());
+
+        $items = $client->selectDatabase(
+            'test',
+            [ 'readPreference' => new \MongoDB\Driver\ReadPreference(\MongoDB\Driver\ReadPreference::RP_SECONDARY) ]
+        )->items;
+
+        foreach ($items->find( [ 'end' => [ '$exists' => false ] ], [ 'session' => $s2 ] ) as $item) {
```

Comment:
It may be more readable to do `find()` on its own (splitting the arguments on their own line) and change this to `foreach ($cursor as $item)`.